File: java/src/Solution_Sword_06.java
import entity.ListNode;

import java.util.ArrayList;

public class Solution_Sword_06 {
    public int[] reversePrint(ListNode head) {
        if (head == null) return new int[0];
        ArrayList<Integer> arrayList = new ArrayList<>();
        while (head != null) {
            arrayList.add(0, head.val);
            head = head.next;
        }
        int[] result = new int[arrayList.size()];
        for (int i = 0; i < arrayList.size(); ++i) {
            result[i] = arrayList.get(i);
        }
        return result;
    }
}

File: python/Solution_Sword_06.py
# !/usr/bin/python
# coding: utf8
# Time: 2020/9/24 上午10:15
# Author: Liam
# E-mail: luyu.real@qq.com
# Software: PyCharm
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
class Solution:
    def reversePrint(self, head):
        result = []
        while head:
            result.insert(0, head.val)
            head = head.next
        return result

